src/nwb_conversion_tools/tools/data_transfers.py

```diff
@@ -84,3 +89,62 @@ def estimate_s3_conversion_cost(
     cost_gb_m = 0.08 / 1e3  # $0.08 / GB Month
     cost_mb_s = cost_gb_m / (1e3 * 2.628e6)  # assuming 30 day month; unsure how amazon weights shorter months?
     return cost_mb_s * total_mb_s
+
+
+def automatic_dandi_upload(
+    nwb_folder_path: FolderPathType,
+    dandiset_id: str,
+    version: Optional[str] = None,
+    staging: bool = False,
+    api_token: Optional[str] = None,
+):
+    """
+    Fully automated upload of NWBFiles to a DANDISet.
+
+    Parameters
+    ----------
+    nwb_folder_path : FolderPathType
+        Folder containing the NWBFiles to be uploaded.
+    dandiset_id : str
+        Six-digit string identifier for the DANDISet the NWBFiles will be uploaded to.
+    version : str, optional
+        "draft" or "version".
+        The default is "draft".
+    staging : bool, optional
+        Is the DANDISet hosted on the staging server? This is mostly for testing purposes.
+        The default is False.
+    api_token : str
+        Your API token for your DANDI account - DO NOT STORE THIS IN ANY CODE ON GITHUB.
+        Use environment variables for CI, or interactivity for personal usage.
+        Or store personal conversion scripts containing the token outside of GitHub.
+    """
+    version = "draft" if version is None else version
+    if api_token is None:
+        api_token = input("Please enter your DANDI API token: ")
+    url_base = "https://gui-staging.dandiarchive.org" if staging else "https://dandiarchive.org"
+    dandiset_url = f"{url_base}/dandiset/{dandiset_id}/{version}"
+
+    os.chdir(nwb_folder_path.parent)
+    validate_return = os.popen(f"dandi validate {nwb_folder_path.name}").read().strip()
+    assert re.fullmatch(
+        pattern=r"^Summary: No validation errors among \d+ file\(s\)$", string=validate_return
+    ), "DANDI validation failed!"
+
+    download_return = os.popen(f"dandi download {dandiset_url}").read()
+    assert download_return, "DANDI download failed!"  # output is a bit too dynamic to regex; if it fails it is empty
+
+    os.chdir(nwb_folder_path.parent / dandiset_id)
+    os.system(f"dandi organize ../{nwb_folder_path.name}")
+    assert len(list(Path.cwd().iterdir())) > 1, "DANDI organize failed!"
+
+    dandi_upload_command = "dandi upload -i dandi-staging" if staging else "dandi upload"
+    proc = Popen(dandi_upload_command, shell=True, stdin=PIPE)
+    proc.stdin.write(bytes(f"{api_token}", "utf-8"))
```

Comment:
It might be easier to set DANDI_API_KEY as an environment variable

Reply:
Agreed, it's really the safest way. We still have to pipe it to the dandi upload prompt however, as simply setting it does not prevent the prompt from happening. But definitely preferable to the `input` above.

Reply:
set, not get. I don't really understand what's happening here.

Reply:
When you run `dandi upload`, it prompts interactively for an API key. We can pass it automatically in the program via the lines above, if we have read it from the system in the first few lines of the function. I'm assuming now that the user has set it as an environment variable prior to running the program.

Reply:
If you set the enviroment variable you won't be prompted for the key

Reply:
See https://www.dandiarchive.org/handbook/10_using_dandi/#storing-access-credentials

Reply:
Ah, I had tried this the python way `import os; os.environ["DANDI_API_KEY"] = "...."` but that did work for me, but setting it in the bash terminal does.

Reply:
Yeah that's apparently a problem of scope.

Within Python instance run inside a bash instance, doing the `os` method only adds it to environment variables inside the Python instance, not inside the outer bash instance.

But setting it at the bash level then using it in Python code called from that same bash instance works fine.

Now instructing users to set it at the bash level before running any code involving this helper function.